server/src/main/java/org/elasticsearch/action/synonyms/GetSynonymsAction.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.synonyms;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.synonyms.SynonymRule;
+import org.elasticsearch.synonyms.SynonymsManagementAPIService;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public class GetSynonymsAction extends ActionType<GetSynonymsAction.Response> {
+
+    public static final GetSynonymsAction INSTANCE = new GetSynonymsAction();
+    public static final String NAME = "cluster:admin/synonyms/get";
+
+    public GetSynonymsAction() {
+        super(NAME, Response::new);
+    }
+
+    public static class Request extends ActionRequest {
+        private static final int MAX_SYNONYMS_RESULTS = 10_000;
+        private final String synonymsSetId;
+        private final int from;
+        private final int size;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.synonymsSetId = in.readString();
+            this.from = in.readVInt();
+            this.size = in.readVInt();
+        }
+
+        public Request(String SynonymsSetId, int from, int size) {
+            Objects.requireNonNull(SynonymsSetId, "Synonym set ID cannot be null");
+            this.synonymsSetId = SynonymsSetId;
+            this.from = from;
+            this.size = size;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            validationException = validatePositiveInt("from", from, validationException);
+            validationException = validatePositiveInt("size", size, validationException);
+
+            if (from + size > MAX_SYNONYMS_RESULTS) {
+                validationException = addValidationError(
+                    "Too many synonyms to retrieve. [from] + [size] must be less than or equal to " + MAX_SYNONYMS_RESULTS,
+                    validationException
+                );
+            }
+
+            return validationException;
+        }
+
+        private ActionRequestValidationException validatePositiveInt(
```

Comment:
nit, optional: should we make this function `static`?

Reply:
Good catch, thanks!